Managers/PhaseManager.gd

```diff
@@ -13,6 +13,8 @@ signal on_event_win
 ## When the player is dead (reduced to 0 health)
 signal on_defeat
 
+## This is a signal to fix a bug with blocking regarding order of turn start effects, might be removed later
+signal before_phase_changed(new_phase: GlobalEnums.Phase, old_phase: GlobalEnums.Phase)
```

Comment:
that seems like it might cause more problems later.
Implementing an actual new phase for block removal instead of playing with signal order in the same phase will be clearer and less error prone

Reply:
I think this can be fine as a temp solution as long as we remove it when the phase system is changed. If I were to add more phases it would need a bit of reworking the battler, which I think is a bit out of scope, as well as the fact that it would be changed immediately in #118. This problem is a bit to complicated and I think it would be better if we leave it as is then wait for #118 to be finished, then we can revisit this. The only issue is that this could get very messy if we forget about it, but as long as we make a new issue, I'm sure we won't. Also, I could also change the new signal name, to make it clear that it's teporary so that no one uses it